pkg/ddl/column.go

```diff
@@ -731,6 +731,9 @@ func (w *worker) doModifyColumnTypeWithData(
 	case model.StateDeleteOnly:
 		// Column from null to not null.
 		if !mysql.HasNotNullFlag(oldCol.GetFlag()) && mysql.HasNotNullFlag(changingCol.GetFlag()) {
+			if d.lease > 0 {
+				delayForAsyncCommit()
```

Comment:
I think it's not needed. This branch will only happen when the modifying column is implemented through creating a new column and removing the old one. When the new column is in the `DeleteOnly` state, it'll have no chance to insert `NULL` values into it :thinking:.